atomic_reactor/plugins/exit_koji_import.py

```diff
@@ -633,3 +504,215 @@ def run(self):
                        json.dumps(build_info, sort_keys=True, indent=4))
 
         return build_id
+
+
+class KojiImportPlugin(KojiImportBase):
+
+    key = PLUGIN_KOJI_IMPORT_PLUGIN_KEY  # type: ignore
+
+    def __init__(self, tasker, workflow,
+                 kojihub=None, url=None,
+                 verify_ssl=True, use_auth=True,
+                 koji_ssl_certs=None, koji_proxy_user=None,
+                 koji_principal=None, koji_keytab=None,
+                 blocksize=None,
+                 target=None, poll_interval=5):
+        super(KojiImportPlugin, self).__init__(tasker, workflow,
+                                               kojihub, url,
+                                               verify_ssl, use_auth,
+                                               koji_ssl_certs, koji_proxy_user,
+                                               koji_principal, koji_keytab,
+                                               blocksize,
+                                               target, poll_interval)
+
+    def set_media_types(self, extra, worker_metadatas):
+        media_types = []
+        for platform in worker_metadatas:
+            annotations = get_worker_build_info(self.workflow,
+                                                platform).build.get_annotations()
+            if annotations.get('media-types'):
+                media_types = json.loads(annotations['media-types'])
+                break
+        super(KojiImportPlugin, self).set_media_types(extra, worker_metadatas,
+                                                      media_types)
+
+    def get_build(self, metadata, worker_metadatas):
+        start_time = int(atomic_reactor_start_time)
+        extra = super(KojiImportPlugin, self)._setup_get_build()
```

Comment:
This seems to be identical for both source and non-source builds, is there a reason why it could not be done in the base `_get_build()` method? You also would not need to pass the `start_time, extra` params to it in that case.

Reply:
AFAICT from local unit testing, `start_time` *has* to be invoked at the head of the method which ends up using it (if that makes sense), otherwise the unit tests of the timestamps fail...